lib/renderror/validate.rb

```diff
@@ -0,0 +1,58 @@
+module Renderror
+  module Validate
+    extend ActiveSupport::Concern
+
+    PERMITTED_VALIDATIONS = %i[
+      jsonapi_type jsonapi_id
+    ].freeze
+
+    included do
+      def validate_jsonapi_type
+        return unless params['data'].present?
+
+        raise Renderror::Conflict unless type_matches?
+      end
+
+      def type_matches?
+        params['data'].try(:[], 'type') == jsonapi_type
+      end
+
+      # This can be overwritten on a per-controller basis if the type name
+      # doesn't match the controller name
```

Comment:
In another update I wouldn't mind changing this to match our other helpers:

```ruby
renderror_validate jsonapi_type: 'users'
```

Reply:
In fact extending on this, maybe rather than adding a `before_action` for custom controller methods, we could have:

```ruby
renderror_validate jsonapi_type: 'users', on: :change_status
```

Reply:
I like that, more explicit 👍

Reply:
Haha yeah. Might do that in another PR. I think this approach is OK for now, given that it's only custom actions where this will come up.